src/plugins/data/public/ui/settings/settings.ts

```diff
@@ -149,8 +155,15 @@ interface Deps {
   search: ISearchStart;
   storage: IStorageWrapper;
   queryEnhancements: Map<string, QueryEnhancement>;
+  searchBarExtensions: SearchBarExtensionConfig[];
 }
 
-export function createSettings({ config, search, storage, queryEnhancements }: Deps) {
-  return new Settings(config, search, storage, queryEnhancements);
+export function createSettings({
+  config,
+  search,
+  storage,
+  queryEnhancements,
+  searchBarExtensions,
+}: Deps) {
+  return new Settings(config, search, storage, queryEnhancements, searchBarExtensions);
```

Comment:
yeah @mengweieric called it [here](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7001#discussion_r1638745574), once your stuff is merged I will convert this to an object as well.

Reply:
i was thinking the same thing, i tried to convert `ConstructorParameters<typeof Settings>` to a map on the type level but typescript isn't that flexible and can't manipulate parameter names